clippy_lints/src/format.rs

```diff
@@ -53,14 +53,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                         // and that the argument is a string
                         if check_arg_is_display(cx, &args[1]);
                         then {
-                            span_lint(cx, USELESS_FORMAT, span, "useless use of `format!`");
+                            let sugg = format!("{}.to_string()", snippet(cx, args[1].span, "<expr>").into_owned());
+                            span_lint_and_then(cx, USELESS_FORMAT, span, "useless use of `format!`", |db| {
+                                db.span_suggestion(expr.span, "consider using .to_string()", sugg);
+                            });
                         }
                     }
                 },
                 // `format!("foo")` expansion contains `match () { () => [], }`
                 ExprMatch(ref matchee, _, _) => if let ExprTup(ref tup) = matchee.node {
                     if tup.is_empty() {
-                        span_lint(cx, USELESS_FORMAT, span, "useless use of `format!`");
+                        let sugg = format!("{}.to_string()", snippet(cx, expr.span, "<expr>").into_owned());
+                        span_lint_and_then(cx, USELESS_FORMAT, span, "useless use of `format!`", |db| {
+                            db.span_suggestion(expr.span, "consider using .to_string()", sugg);
```

Comment:
Same here, just with `span`

Reply:
I changed the `span` on line 67 to `expo.span`. This is what it outputs now:
```
error: useless use of `format!`
 --> $DIR/format.rs:6:13
  |
6 |     format!("foo");
  |             ^^^^^ help: consider using .to_string(): `"foo".to_string()`
  |
  = note: `-D useless-format` implied by `-D warnings`

error: aborting due to previous error
```
This is what it output before:
```
error: useless use of `format!`
 --> $DIR/format.rs:6:5
  |
6 |     format!("foo");
  |     ^^^^^^^^-----^^
  |             |
  |             help: consider using .to_string(): `"foo".to_string()`
  |
  = note: `-D useless-format` implied by `-D warnings`

error: aborting due to previous error
```
I personally think that the arrows under the whole thing (`format!("foo")`), and then the underscores under specifically `"foo"` was better, making it clear the the `format!` macro was the problem, and it would be better if you used `.to_string()` on `"foo"`.

But, I also don't know what the conventions for the errors are, so I am fine if we prefer it the way you suggested.

Reply:
No I meant the other way around. Use `span` instead of `expr.span` also in line 68. The suggestion is automatically applicable by rls and needs to hilight the entire span that should be replaced. The way you changed it now would result in `format!("foo".to_string())`